### PR TITLE
Attempted fix to Travis config for shellcheck

### LINKED ISSUE #310: Travis build checks failing on PRs while attempting to install shellcheck
Public key error.
Blocking https://github.com/AdoptOpenJDK/openjdk-build/pull/308 and https://github.com/AdoptOpenJDK/openjdk-build/pull/309
```
10.84s$ sudo -E apt-get -yq --no-install-suggests --no-install-recommends --force-yes install shellcheck
Reading package lists...
Building dependency tree...
Reading state information...
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:
The following packages have unmet dependencies:
 libc6-dev : Breaks: binutils (< 2.26) but 2.24-5ubuntu14.2 is to be installed
E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.
apt-get.diagnostics
apt-get install failed
$ cat ~/apt-get-update.log
Ign:1 http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.4 InRelease
Ign:3 http://dl.google.com/linux/chrome/deb stable InRelease
Hit:4 http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.4 Release
Hit:5 http://security.ubuntu.com/ubuntu trusty-security InRelease
Get:6 http://ftp.us.debian.org/debian unstable InRelease [242 kB]
Ign:2 http://toolbelt.heroku.com/ubuntu ./ InRelease
Hit:7 http://toolbelt.heroku.com/ubuntu ./ Release
Hit:8 http://dl.google.com/linux/chrome/deb stable Release
Hit:11 http://ppa.launchpad.net/chris-lea/redis-server/ubuntu trusty InRelease
Ign:12 http://archive.ubuntu.com/ubuntu trusty InRelease
Hit:13 https://dl.hhvm.com/ubuntu trusty InRelease
Get:10 http://dl.bintray.com/apache/cassandra 39x InRelease [3,168 B]
Hit:14 http://apt.postgresql.org/pub/repos/apt trusty-pgdg InRelease
Hit:15 http://archive.ubuntu.com/ubuntu trusty-updates InRelease
Ign:17 http://ppa.launchpad.net/couchdb/stable/ubuntu trusty InRelease
Hit:18 http://archive.ubuntu.com/ubuntu trusty-backports InRelease
Get:19 http://ftp.us.debian.org/debian unstable/main amd64 Packages [10.7 MB]
Hit:20 http://archive.ubuntu.com/ubuntu trusty Release
Hit:22 http://ppa.launchpad.net/git-core/ppa/ubuntu trusty InRelease
Get:23 http://ftp.us.debian.org/debian unstable/main i386 Packages [10.7 MB]
Hit:24 http://ppa.launchpad.net/openjdk-r/ppa/ubuntu trusty InRelease
Get:25 http://ftp.us.debian.org/debian unstable/contrib amd64 Packages [75.4 kB]
Get:26 http://ftp.us.debian.org/debian unstable/contrib i386 Packages [70.8 kB]
Get:27 http://ftp.us.debian.org/debian unstable/non-free amd64 Packages [98.7 kB]
Get:28 http://ftp.us.debian.org/debian unstable/non-free i386 Packages [88.9 kB]
Hit:29 http://ppa.launchpad.net/pollinate/ppa/ubuntu trusty InRelease
Hit:30 http://ppa.launchpad.net/webupd8team/java/ubuntu trusty InRelease
Hit:31 https://packagecloud.io/computology/apt-backport/ubuntu trusty InRelease
Hit:32 http://ppa.launchpad.net/couchdb/stable/ubuntu trusty Release
Hit:33 https://packagecloud.io/github/git-lfs/ubuntu trusty InRelease
Hit:35 https://packagecloud.io/rabbitmq/rabbitmq-server/ubuntu trusty InRelease
Fetched 22.0 MB in 2s (8,598 kB/s)
Reading package lists...
W: There is no public key available for the following key IDs:
7638D0442B90D010
W: http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/Release.gpg: Signature by key 15866BAFD9BCC4F3C1E0DFC7D69548E1C17EAB57 uses weak digest algorithm (SHA1)
The command "sudo -E apt-get -yq --no-install-suggests --no-install-recommends --force-yes install shellcheck" failed and exited with 100 during .
Your build has been stopped.
```

### PR DESCRIPTION
**Pull request by karianna**
Attempt to fix #310

### CODE REVIEW

**Comment by gdams** [comment]
Can you fix up the travis failures now?

**Comment by karianna** [comment]
Downside to this style of shellcheck install is that it builds shellcheck from source each time which is not a quick process.

File: git-hg/updateforesttags.sh [hunk 1]
```diff
@@ -60,91 +61,91 @@ if ! which git-remote-hg 2>/dev/null; then
 fi
 
 # Clone current AdoptOpenJDK repo
-cd $WORKSPACE/$GITHUB_REPO
-echo "Clone current $GITHUB_REPO"
-git clone $GITHUB_PROJECT/$GITHUB_REPO.git
-cd $GITHUB_REPO
+cd "$WORKSPACE"/"$GITHUB_REPO" || exit
```
Comment (sxa555) [inline git-hg/updateforesttags.sh]: I'd stick a non-zero return code on here - so `exit 1` (Same for a few other currrences later on)

File: git-hg/updateforesttags.sh [hunk 2]
```diff
@@ -60,91 +61,91 @@ if ! which git-remote-hg 2>/dev/null; then
...
 
 # Clone current openjdk from Mercurial
 echo "Get base openjdk repository" 
-cd $WORKSPACE/openjdk/mirror
+cd "$WORKSPACE"/openjdk/mirror || exit
 git init
 git clone --bare "hg::${HG_REPO}"
 
-cd "$OPENJDK_VERSION.git"
+cd "$OPENJDK_VERSION.git" || exit
```
Comment (sxa555) [inline git-hg/updateforesttags.sh]: This is slightly different from the changes you've made earlier - compare "cd "$OPENJDK_VERSION".git` and `cd $OPENJDK_VERSION.git"` (although granted this is just one level)

I tend to prefer using quotes around the whole thing as it looks a but cleaner so perhaps use things like `cd "$WORKSPACE/openjdk/mirror"` in all cases?

File: git-hg/updateforesttags.sh [hunk 3]
```diff
@@ -60,91 +61,91 @@ if ! which git-remote-hg 2>/dev/null; then
...
 
 git filter-branch -f --index-filter 'git rm -r -f -q --cached --ignore-unmatch .hg .hgignore .hgtags get_source.sh' --prune-empty --tag-name-filter cat -- --all
 
-cd ..
+cd .. || exit
```
Comment (sxa555) [inline git-hg/updateforesttags.sh]: I would hope that one can't go wrong ;-)

File: git-hg/updateforesttags.sh [hunk 4]
```diff
@@ -20,38 +20,39 @@
...
    jdk9*) HG_REPO=http://hg.openjdk.java.net/jdk_updates/jdk9u
-          [ -z "$TAGS" ] && TAGS="jdk-9+181 jdk-9.0.1+11 jdk-9.0.3+9 jdk-9.0.4+11" ;;
+          [ -z "$TAGS" ] && TAGS="jdk-9+181 jdk-9.0.1+11 jdk-9.0.3+9 jdk-9.0.4+11";;
        *) Unknown JDK version - only jdk8u and jdk9 are supported; exit 1;;
 esac
 
 # Clean up
-rm    -rf $WORKSPACE/$GITHUB_REPO $WORKSPACE/openjdk
-mkdir -p  $WORKSPACE/$GITHUB_REPO $WORKSPACE/openjdk/mirror 
+rm    -rf "${WORKSPACE}"/"${GITHUB_REPO}" "${WORKSPACE}"/openjdk
```
Comment (sxa555) [inline git-hg/updateforesttags.sh]: Perhaps change other instances to use `{}` (or remove them from this one) to be consistent?

File: git-hg/updateforesttags.sh (hunk 1 above)
Comment (karianna) [reply git-hg/updateforesttags.sh]: Agreed, changing...

File: git-hg/updateforesttags.sh (hunk 2 above)
Comment (karianna) [reply git-hg/updateforesttags.sh]: Agreed, changing...

**Pull request merged**